rotkehlchen/tests/api/test_custom_assets.py

```diff
@@ -0,0 +1,210 @@
+        ),
+    )
+    result = assert_proper_response_with_result(response)
+    assert result['entries'] == [entry.to_dict() for entry in generated_entries]
+
+    # test that filtering works as expected
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json={'name': 'Gold Bar'},
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result['entries']) == 1
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json={'custom_asset_type': 'real estate'},
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result['entries']) == 2
+
+    # test that filtering with an invalid value fails
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json={'name': 'idontexist'},
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result['entries']) == 0
+
+
+def test_add_custom_asset(rotkehlchen_api_server) -> None:
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json={'name': 'XYZ', 'custom_asset_type': 'stocks'},
+    )
+    assert_proper_response(response)
+
+    # test that adding custom asset with an existing name fails
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json={'name': 'XYZ'},
+    )
+    assert_error_response(
+        response,
+        contained_in_msg='Failed to add custom asset as entry with identifier',
+        status_code=HTTPStatus.CONFLICT,
+    )
+
+    # test that adding a custom asset without a name fails
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json={'custom_asset_type': 'rah'},
+    )
+    assert_error_response(response, contained_in_msg='Missing data for required field')
+
+
+def test_edit_custom_asset(rotkehlchen_api_server) -> None:
+    generated_entries = _populate_custom_assets_table(rotkehlchen_api_server.rest_api.rotkehlchen.data.db)  # noqa: E501
+    data = generated_entries[0].to_dict()
+    data['name'] = 'Milky Way'
+    response = requests.patch(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json=data,
+    )
+    assert_proper_response(response)
+
+    # check that the asset was indeed updated
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'customassetsresource',
+        ),
+        json={'name': 'Milky Way'},
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result['entries']) == 1
```

Comment:
Maybe we could compare the result here with `data` to get more thorough check?

(same comment for other checks in this test)